networkx/algorithms/approximation/steinertree.py

```diff
@@ -0,0 +1,76 @@
+import networkx as nx
+from itertools import combinations, chain
+from networkx.utils import pairwise, not_implemented_for
+
+__all__ = ['metric_closure', 'steiner_tree']
+
+
+@not_implemented_for('directed')
+def metric_closure(G, nodes, weight="weight"):
```

Comment:
As I mentioned in the discussion page for this pull request, I think this should be just `def metric_closure(G, weight='weight'):`, and you should just return the metric closure of the entire graph unless there is a good reason to do this optimization.